packages/stylelint-plugin/src/rules/no-invalid-functions.ts

```diff
@@ -0,0 +1,360 @@
+  if (percentageOrDecimal.indexOf("%") !== -1) {
+    return parsePercentage(percentageOrDecimal)
+  } else {
+    const parsed = parseFloat(percentageOrDecimal)
+    if (isNaN(parsed)) {
+      return "NaN"
+    } else {
+      return parsed
+    }
+  }
+}
+
+/**
+ * Given a string like "50%", "40%", "30", "15" etc (you don't have to pass in the percentage sign), return a decimal like 0.5, 0.4, 0.3 etc
+ */
+const parsePercentage = (value: string): number | "NaN" => {
+  // Add-alpha can be used without a percentage sign, that won't fly.
+  // Fix the percentage sign, then convert to a decimal for usage in RGBA
+  const parsed = parseInt(value.replace(/\%/, ""), 10)
+  if (isNaN(parsed)) {
+    return "NaN"
+  }
+  return parsed / 100
+}
+
+/*
+  Return a replacement rgb-params variable
+*/
+const getAndReportOnReplacementRgbParamsVariable = (
+  variable: Variable,
+  decl: Declaration,
+  options: Options
+) => {
+  const variablePrefix = variablePrefixForLanguage(options.language)
+  if (!variable.kaizenToken) return undefined
+
+  if (!variable.kaizenToken.cssVariable) {
+    const fixedVariable =
+      kaizenTokensByName[
+        variable.kaizenToken.name.replace("kz", "kz-var") + "-rgb-params"
+      ]
+    if (!fixedVariable) {
+      options.reporter({
+        message: noMatchingRgbParamsVariableMessage(variable.kaizenToken.name),
+        node: decl,
+        autofixAvailable: false,
+      })
+      return undefined
+    }
+    if (options.fix && !isVariable(decl)) {
+      return `${variablePrefix}${fixedVariable.name}`
+    } else {
+      options.reporter({
+        message: invalidRgbaUsage(fixedVariable.name),
+        node: decl,
+        autofixAvailable: !isVariable(decl),
+      })
+    }
+  }
+
+  // If the value is not like `123, 123, 123`, AND it's a CSS variable, it's not valid
+  if (
+    variable.kaizenToken.cssVariable &&
+    variable.kaizenToken.cssVariable.fallback &&
+    !isRgbTriple(variable.kaizenToken.cssVariable.fallback)
+  ) {
+    const fixedVariable =
+      kaizenTokensByName[variable.kaizenToken.name + "-rgb-params"]
+    if (!fixedVariable) {
+      options.reporter({
+        message: noMatchingRgbParamsVariableMessage(variable.kaizenToken.name),
+        node: decl,
+        autofixAvailable: false,
+      })
+      return undefined
+    }
+    if (options.fix && !isVariable(decl)) {
+      return `${variablePrefix}${fixedVariable.name}`
+    } else {
+      options.reporter({
+        message: invalidRgbaUsage(fixedVariable.name),
+        node: decl,
+        autofixAvailable: !isVariable(decl),
+      })
+    }
+  }
+}
+
+/**
+ * This function will compile a value, for example `mix($kz-color-wisteria-800, $white, 80%)` in place (which would result in #5d5f6e).
+ * It also uses the variables that are immediately available within the same stylesheet, so nothing too fancy, and will bail out with a lint error if it can't compile.
+ * ALSO, it will work with $kz-var-* variables too.
```

Comment:
I think I made this comment elsewhere, but I'm uncomfortable with this behaviour. I think leaving the static sass variables (that contain actual values, not references to css variables) is safer than compiling hard coded values.